pstree.go

```diff
@@ -92,17 +92,25 @@ type ProcessStat struct {
 }
 
 func scan(dir string) (Process, error) {
-	f, err := os.Open(filepath.Join(dir, "stat"))
+	stat := filepath.Join(dir, "stat")
+	data, err := ioutil.ReadFile(stat)
 	if err != nil {
 		// process vanished since Glob.
 		return Process{}, nil
 	}
-	defer f.Close()
+	info := strings.FieldsFunc(string(data), func(r rune) bool {
+		return r == '(' || r == ')'
+	})
+
+	if len(info) != 3 {
+		return Process{}, fmt.Errorf("file %s format not correct", stat)
+	}
 
 	var proc Process
-	_, err = fmt.Fscanf(
-		f, statfmt,
-		&proc.Stat.Pid, &proc.Stat.Comm, &proc.Stat.State,
+	proc.Stat.Comm = info[1]
+	_, err = fmt.Sscanf(
+		info[0]+info[2], statfmt,
```

Comment:
instead of concatenating those 2 strings, perhaps just use `strconv.ParseInt` for `info[0]` and use `fmt.Sscanf` for `info[2]` ?

Reply:
strconv.ParseInt and  fmt.Sscanf will have two errors to deal with, while concatenating just have one error to deal with , i am lazy :-)